terraform/modules/iam/iam.tf

```diff
@@ -0,0 +1,111 @@
+resource "aws_iam_role" "ecs_task_execution_role" {
+  name               = var.ecs_task_execution_role_name
+  assume_role_policy = data.aws_iam_policy_document.assume_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_task_execution_role_policy_attachment" {
+  role       = aws_iam_role.ecs_task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_policy" "ecr_policy" {
+  name        = var.ecs_ecr_policy_name
+  description = "Policy for ECS tasks to access ECR"
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "ecr:GetAuthorizationToken",
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:BatchGetImage"
+        ]
+        Resource = [
```

Comment:
Do we need to add the `message-parser` and `trigger code reference` to this iam policy?

Reply:
Hmm, seems like we don't because `ecs` is successfully pulling the images from `ecr`. I'll fiddle with it to see what's up!

Reply:
I refactored and correctly set the task execution role and the task role on the task definition, the changes for that are:

- execution role: this inherits from a preexisting AWS resource which allows it access to all the things that it needs to provision the container.
- task role: this policy is mostly blank except for the s3 bucket that needs to be accessible by the ecr-viewer

I'm going to be working with a member of the dev team to work out if this is acceptable but it's pretty well cut down on open permissions.